libs/serviceproxy.py

```diff
@@ -131,8 +132,27 @@ def getEvent(self):
                 event = jsonhelpers.decode(servicedefs.DiagnosticEvent, evJsonStr)
         return event
 
-
     def saveto(self, path, alternatePath):
         req = servicedefs.SavetoRequest(self.incrSeq(), servicedefs.ReloadRequestArgs(path, alternatePath))
         jsonStr = jsonhelpers.encode(req)
         self.__comm.postCmd(jsonStr)        
+
+    def navTo(self, searchText, fileName):
+        req = self.wrap_req_dict ({
+            'command': 'navto',
+            'arguments': {
+                'searchValue': searchText,
+                'file': fileName,
+                'maxResultCount': 20
+            }
+        })
+        resp_str = self.__comm.sendCmdSync(json.dumps(req), req['seq'])
+        if resp_str['success']:
+            return resp_str['body']
+        else:
+            return {}
+
+    def wrap_req_dict(self, request_dict):
```

Comment:
As we are moving to remove the whole object definition layer, I want to try using the dictionary pattern for nav to command. And I made the function because I figure the adding sequence id and message type part can be reused by other methods.